examples/robot.py

```diff
@@ -0,0 +1,128 @@
+        self.gripper_link_multiplier: List[float, ...] = [-1, -1, 1, 1, 1, -1]
+        self.gripper_link_limit: List[float, float] = [0, 1.343904]
+        self.gripper_x_line = None
+        self.gripper_y_line = None
+        self.gripper_z_line = None
+
+        for i in range(len(self.gripper_link_ids)):
+            if i != 0:
+                c = bc.createConstraint(self.robot_id, self.gripper_link_ids[0],
+                                        self.robot_id, self.gripper_link_ids[i],
+                                        jointType=bc.JOINT_GEAR,
+                                        jointAxis=[0, 1, 0],
+                                        parentFramePosition=[0, 0, 0],
+                                        childFramePosition=[0, 0, 0])
+                gearRatio = -self.gripper_link_multiplier[0] * self.gripper_link_multiplier[i]
+                bc.changeConstraint(c, gearRatio=gearRatio, maxForce=3, erp=1)
+
+            gripper_link_limit = sorted([limit * self.gripper_link_multiplier[i] for limit in self.gripper_link_limit])
+            print(gripper_link_limit)
+            bc.changeDynamics(self.robot_id, self.gripper_link_ids[i],
+                              jointLowerLimit=gripper_link_limit[0],
+                              jointUpperLimit=gripper_link_limit[1])
+
+            # Initially set the gripper as opened
+            bc.resetJointState(self.robot_id, self.gripper_link_ids[i],
+                               self.gripper_link_limit[1] * self.gripper_link_multiplier[i])
+
+    def reset_arm_poses(self):
+        for rest_pose, joint_id in zip(self.arm_rest_poses, self.arm_joint_ids):
+            self.bc.resetJointState(self.robot_id, joint_id, rest_pose)
+
+    def open_gripper(self):
+        self.bc.setJointMotorControlArray(self.robot_id, self.gripper_link_ids, self.bc.POSITION_CONTROL,
+                                          targetPositions=[i * self.gripper_link_limit[1] for i in
+                                                           self.gripper_link_multiplier],
+                                          positionGains=[1 for i in range(len(self.gripper_link_ids))])
+
+    def close_gripper(self):
+        # Releasing the position control, leaving a small friction
+        self.bc.setJointMotorControlArray(self.robot_id, self.gripper_link_ids, self.bc.VELOCITY_CONTROL,
+                                          forces=[0.1 for i in range(len(self.gripper_link_ids))])
+        # Force/torque control
+        self.bc.setJointMotorControl2(self.robot_id, self.gripper_link_ids[0], self.bc.TORQUE_CONTROL,
+                                      force=-3 * self.gripper_link_multiplier[0])
+
+    def draw_gripper_frame(self):
+        scale: float = 0.1
+        ee_state = np.array(self.bc.getLinkState(self.robot_id, 6))
+        ee_rotation_matrix = np.array(self.bc.getMatrixFromQuaternion(ee_state[1])).reshape((3, 3))
+        gripper_state = ee_state[0] + ee_rotation_matrix[:, 2] * 0.221
```

Comment:
Just to avoid magic constants:
```suggestion
# Somewhere...
# GRIPPER_Z_OFFSET:float = 0.221
        gripper_pos = ee_state[0] + ee_rotation_matrix[:, 2] * GRIPPER_Z_OFFSET
```